src/python/pants/source/source_root.py

```diff
@@ -93,6 +100,12 @@ def find_by_path(self, path):
       # TODO: Remove this logic. It should be an error to have no matching source root.
       return SourceRoot(path, [], SourceRootCategories.UNKNOWN)
 
+  def traverse(self) -> Set[str]:
+    return self._trie.traverse()
+
+  def trie_find(self, dir_path: str) -> SourceRoot:
```

Comment:
The fact that we use a trie is an implementation detail, so this should have a different name. Ideally one that explains what this does and how it's different from `find_by_path`...

Also, let's add docstrings to all the new methods.

Reply:
I think I actually duplicated `find_by_path`'s functionality with `trie_find`, so I'll just remove this and use that instead.